pkg/agent/client/client.go

```diff
@@ -465,3 +466,20 @@ func (c *client) newAgentClient(ctx context.Context) (agentv1.AgentClient, *node
 	c.connections.AddRef()
 	return c.createNewAgentClient(c.connections.conn), c.connections, nil
 }
+
+// withErrorFields add fields of gRPC call status in logger
+func (c *client) withErrorFields(err error) logrus.FieldLogger {
+	if err == nil {
+		return c.c.Log
+	}
+
+	logger := c.c.Log.WithError(err)
+	if s, ok := status.FromError(err); ok {
+		logger = c.c.Log.WithFields(logrus.Fields{
```

Comment:
logger variable is not used to add fields, and setted with a new one, so `WithError` is missed, 
you must use logger.WithFields instead

Reply:
Fixed at [7b9fc68](https://github.com/spiffe/spire/pull/4262/commits/7b9fc681670652ed020d29f652e02a8c8fbf530f)